examples/sprite_sheets/combine_images_into_sprite_sheet/combine_images_into_sprite_sheet.lpr

```diff
@@ -1,52 +1,118 @@
-
   "Castle Game Engine" is free software; see the file COPYING.txt,
   included in this distribution, for details about the copyright.
-
   "Castle Game Engine" is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-
   ----------------------------------------------------------------------------
 }
 
-{ Generate sprite sheet from a sequence of images.
+{ Generate sprite sheet from a sequence of images, it can contains more then one animation
+  (Images must all have the same size and same name and must be renamed sequentially as follows:
+  img_0, img_1, img_2 ... img_29
+  or
+  img_1, img_2, img_3 ... img_30
+  or
+  img_00 ... img_29
+  or
+  img_001 ... img_030
+  and so on
+  First image must always begin with 0 or 1
+  If the sprite sheet is to contain multiple animations, they must still be renamed in the same way
+  )
+  Generate .xml file related
+  (.xml file can be used e.g. with the sprite-sheet-to-x3d tool to load animation into TCastleScene)
+  .xml file can have the coordinate (0,0) set at the top left or bottom left of the spritesheet
+  (according to your needs)
 
   Use from the command-line:
-  - 1st parameter is the image sequence URL (filename),
-    using @counter(xxx),
-    like https://castle-engine.io/x3d_implementation_texturing_extensions.php#section_ext_movie_from_image_sequence
+  - 1st parameter is the first image sequence URL (filename),
     It can also point to a movie file, like avi or mpg, if you have ffmpeg installed.
-  - 2nd parameter is output image name
-  - 3rd parameter is the number of columns (how many frames in a row)
 
+  - 2nd parameter is 0 or a string containing the animation parameters
   For example:
-  ./combine_images_into_sprite_sheet walking_@counter(1).png sprite_sheet_24fps.png 6
+  Type 0 if your spritesheet contains only one animation
+  Type a string as below if spritesheet contains multiple animations:
+  walk,0,9-run,10,19-idle,20,29
+  or
+  walk,1,10-run,11,20-idle,21,30
+  where the first parameter will be the name of the animation, the second and third parameter
+  the index of the first and last image of the animation
+  Each animation must be separated by the "-"
+  In the first case, only one animation, it will take the same name as the image (without "_x")
+  If you want to give the animation a different name instead of using the parameter "0" 
+  use only one set of parameters as in the case of multiple animations
+  e.g. with "walking_1.png" as the first image and 30 images in total: NewAnimName,1,30
+  
+
+  - 3rd parameter is the is the padding you used
+  For example:
+  Type 1 if the first image uses the form _0 or _1
+  Type 3 if the first image uses the form _000 or _001
+
+  - 4nd parameter is output image name
+
+  - 5rd parameter is the number of columns (how many frames in a row)
+
+  - 6rd parameter is 1 or 0 (True or False).
```

Comment:
Instead of a required 6th parameter with 0 or 1, instead handle an optional parameter like `--reverse-y`. To handle optional parameters, use `CastleParameters` feature to parse command-line by `Parameters.Parse(Options, @OptionProc, nil)` -- example of this is in various CGE tools, like https://github.com/castle-engine/castle-engine/blob/master/tools/texture-font-to-pascal/texture-font-to-pascal.lpr .